File: 1-arrays/217-contains-duplicate.cpp
#include <bits/stdc++.h>
using namespace std;

bool containsDuplicate(vector<int>& nums) {
    unordered_set<int> s;
    for(int num: nums){
        if(!s.insert(num).second)
            return true;
        /* if(s.find(num) != s.end()){
            return true;
        }
        s.insert(num); */
    }
    return false;
}

int main(){
    vector<int> nums = {1,2,3,4,1};
    cout <<boolalpha <<containsDuplicate(nums) <<endl;
}

// Time Complexity: O(n)
// Space Complexity: O(n)

File: 1-arrays/java/217-contains-duplicate.java
import java.util.HashSet;
public class solution{
    public static boolean containsDuplicate(int[] nums) {
        HashSet<Integer> numbers = new HashSet<>();
        for(int num: nums){
            if(numbers.contains(num)){
                return true;
            }
        numbers.add(num);
        }
        return false;
    }
    public static void main(String args[]){
        int nums[] = {1,2,3,4,1};

        System.out.println(containsDuplicate(nums));
    }
}

// Time Complexity: O(n)
// Space Complexity: O(n)
